### PR TITLE
[WIP] DD4hep BField in Pandora

### PR DESCRIPTION
**Pull request by rete**
Fixes [iLCSoft/ILDConfig#58](https://github.com/iLCSoft/ILDConfig/issues/58)
BEGINRELEASENOTES
- New class DDBFieldPlugin:
   - Returns the B field magnitude from DD4hep detector field to Pandora
- DDPandoraPFANewProcessor:
   - Added optional registration of DDBFieldPlugin under processor parameter condition "UseDD4hepField" (default false)
- DDCaloDigi:
   - Fixed warning

ENDRELEASENOTES

### CODE REVIEW

**Comment by rete** [comment]
WIP : I need to validate that with a constant field map the results are the same as the current settings

**Review by gaede — COMMENTED** [review]
Are you sure that returning the magnitude of the B-field is correct here:
https://github.com/rete/DDMarlinPandora/blob/e2f00a897746e58a362349f369272dc81e5901a2/src/DDBFieldPlugin.cc#L26
Should it not be the z-component only ?

**Comment by rete** [comment]
I was not really sure. It also depends on how pandora use it internally for its calculations. I have to look in LCContent

File: src/DDBFieldPlugin.cc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{
+    /* nop */
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+float DDBFieldPlugin::GetBField(const pandora::CartesianVector &positionVector) const
+{
+    double bfield[3] = {0.};
+    m_field.magneticField({positionVector.GetX(), positionVector.GetY(), positionVector.GetZ()}, bfield);
```
Comment (andresailer) [inline src/DDBFieldPlugin.cc]: I think you need to do unit conversion for X,Y,Z

Comment (rete) [reply src/DDBFieldPlugin.cc]: Done. Is this correct ?

**Comment by rete** [comment]
@gaede : I suggest to merge it now for the head version installation. I would like to test it.

**Pull request merged**